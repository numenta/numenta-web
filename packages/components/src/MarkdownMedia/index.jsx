// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import {getVideoIdFromUrl} from 'numenta-web-shared-utils/lib/universal'
import Image from 'numenta-web-shared-components/lib/Image'
import React from 'react'
import Sound from 'numenta-web-shared-components/lib/Sound'
import Video from 'numenta-web-shared-components/lib/Video'

import styles from './index.css'


/**
 * Markdown Media item wrapper for lone Markdown images, sound, video, etc. -
 *  React view component.
 */
const MarkdownMedia = ({border, markdown, shadow}) => {
  const {hideImage, image, sound, time, title, video} = markdown
  const heading = title || 'Numenta Media'
  let media

  if (image && !hideImage) {
    if (video) {
      media = (
        <Video
          border={border}
          image={image}
          respond="mw"
          shadow={true}
          time={time}
          title={heading}
          type="youtube"
          videoId={getVideoIdFromUrl(video)}
        />
      )
    }
    else if (sound) {
      media = (
        <Sound
          border={border}
          image={image}
          respond="mw"
          shadow={true}
          title={heading}
          type="soundcloud"
          url={sound}
        />
      )
    }
    else {
      media = (
        <Image
          alt={heading}
          border={border}
          respond="mw"
          shadow={shadow}
          src={image}
        />
      )
    }
  }

  return (
    <div className={styles.markdownMedia}>
      {media}
    </div>
  )
}

MarkdownMedia.propTypes = {
  border: React.PropTypes.bool,
  shadow: React.PropTypes.bool,
  markdown: React.PropTypes.object.isRequired,
}

MarkdownMedia.defaultProps = {
  border: true,
  shadow: true,
}

export default MarkdownMedia
