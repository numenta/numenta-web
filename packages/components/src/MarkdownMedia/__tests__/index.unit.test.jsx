import React from 'react'
import renderer from 'react-test-renderer'

import MarkdownMedia from '../../MarkdownMedia'

const media = {
  title: 'Test',
  image: 'test.png',
  hideImage: false,
  sound: false,
  video: false,
}


describe('MarkdownMedia React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <MarkdownMedia markdown={media} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
