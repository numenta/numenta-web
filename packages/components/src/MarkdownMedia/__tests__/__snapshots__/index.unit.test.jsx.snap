exports[`MarkdownMedia React component Renders correctly 1`] = `
<div
  className="markdownMedia">
  <img
    alt="Test"
    className="image border mw shadow"
    onClick={undefined}
    src="test.png"
    title={undefined} />
</div>
`;
